controllers/selfnoderemediationconfig_controller.go

```diff
@@ -190,3 +198,54 @@ func (r *SelfNodeRemediationConfigReconciler) syncCerts(cr *selfnoderemediationv
 	}
 	return nil
 }
+
+func (r *SelfNodeRemediationConfigReconciler) updateDsTolerations(objs []*unstructured.Unstructured, tolerations []corev1.Toleration) error {
+	r.Log.Info("Updating DS tolerations")
+	if len(tolerations) == 0 {
+		return nil
+	}
+	for _, toleration := range tolerations {
+		for _, ds := range objs {
+			if err := r.updateTolerationOnDs(ds, toleration); err != nil {
```

Comment:
- why iterating the tolerations, and not adding all at once?
- just to protect our future selfs from surprises in case we add manifests to `/install`: we know that there is only 1 object today, the daemonset. What about ensuring that it is really 1 object only, and process it directly without iteration? And fail if there are more manifests, which means this code needs an update?

Reply:
>why iterating the tolerations, and not adding all at once?

IIUC each toleration needs to be converted from `v1.Toleration` to `map[string]interface{}`

>just to protect our future selfs from surprises in case we add manifests to /install: we know that there is only 1 object today, the daemonset. What about ensuring that it is really 1 object only, and process it directly without iteration? And fail if there are more manifests, which means this code needs an update?

Good idea :+1:

Reply:
> is there a simple way to convert the whole slice ?

probably not, but you can have that loop where you do that conversion IMHO

Reply:
If it's not something you feel strongly about I think I prefer to keep as is.
Since we need to iterate over the tolerations in any case I do prefer the pattern of having one method that handles the whole slice which calls a method that handle a single element (toleration).
IMO it's improves readability and is more modular.

Reply:
It doesn't make a difference on performance in this case, but what kind of pattern is it where you do more work than needed inside the loop? (processing the whole DS object vs just converting the tolerations to unstructured) 🤔 

> IMO it's improves readability

let's not start arguing about this 😎 

>  and is more modular.

if you want to have this more modular, put the conversion of the toleration slice into its own function?

Reply:
> but what kind of pattern is it where you do more work than needed inside the loop? 

Fair point

>if you want to have this more modular, put the conversion of the toleration slice into its own function?

Sure, that WFM :+1: